### ISSUE TITLE
@PolyDet("up") should be converted @PolyDet if type can't be @OrderNonDet

### ISSUE THREAD

**Issue by jwaataja** [t-rasmud/checker-framework#96]
Consider the following class:
```java
import org.checkerframework.checker.determinism.qual.*;
import java.util.*;

public class C {
    public static @PolyDet boolean f(@PolyDet C c1, @PolyDet C c2) {
        return c1.equals(c2);
    }
}
```

This method should type check because the determinism of `c1.equals(c2)` should be the same as `c1` and `c2`, which is `@PolyDet`. I get the following error

```
.java:6: error: [return.type.incompatible] incompatible types in return.
        return c1.equals(c2);
                        ^
  type of expression: @PolyDet("up") boolean
  method return type: @PolyDet boolean
```

The type `@PolyDet("up")` doesn't make sense because `C` can't be `@OrderNonDet`. The checker should recognize this and automatically convert it to `@PolyDet`.

**Comment by smillst** [comment]
Is it enough to just change the JDK annotation on the return type of Object#equals to `@PolyDet`?

**Comment by jwaataja** [comment]
It has to be `@PolyDet("up")` to allow for the possibility the actual type is a `Collection`. We made this change because of #86.

**Comment by smillst** [comment]
I can't fix this issue quickly, but I can reduce the related warning suppressions in Randoop:

- 1 warning suppression can be removed by annotating a parameter. (https://github.com/jwaataja/randoop/pull/6)
- 4 warning suppressions can be turned into 1 by by overriding `equals` in `Types`. (https://github.com/jwaataja/randoop/pull/6)
- last 5 warning suppressions can be eliminated by annotated `equals` in Class.  (I'm working on this change now.)

**Comment by t-rasmud** [comment]
Fixed in ee31b95
